pkg/promqlgen/promqlgen.go

```diff
@@ -64,3 +61,55 @@ func LabelSelectorsToRelabelConfig(matches []string) (*monv1.RelabelConfig, erro
 		Regex:        regex,
 	}, nil
 }
+
+func GroupLabelSelectors(matches []string) (string, error) {
+	labelSets, err := parseMetricSelectorFromArray(matches)
+	if err != nil {
+		return "", nil
+	}
+	newLabelSet := map[string][]string{}
+	for _, ls := range labelSets {
+		for _, lm := range ls {
+			_, exists := newLabelSet[lm.Name]
+			if exists {
+				newLabelSet[lm.Name] = append(newLabelSet[lm.Name], lm.Value)
+			} else {
+				newLabelSet[lm.Name] = []string{lm.Value}
+			}
+		}
+
+	}
+
+	keys := make([]string, 0, len(newLabelSet))
+	for k := range newLabelSet {
+		keys = append(keys, k)
+	}
+	sort.Strings(keys)
+
+	res := "{"
+	i := 0
+	for _, k := range keys {
+		res += k + `=~"`
+		res += strings.Join(newLabelSet[k], "|")
+		i++
+		if i == len(newLabelSet) {
```

Comment:
wouldn't it be more correct to say that we add the trailing `|` unless `k == "__name__"`?

Reply:
hmm good point, but what if we have multiple label keys?

Reply:
hmm okay, yes I see what you mean, it just needs one so it will always work, applying suggestion.